### PR TITLE
OSDOCS1834: Adding Ingress Global Access config topic to UPI/IPI GCP install

### PR DESCRIPTION
**Pull request by skrthomas**
Adding Ingress Global Access config topic to UPI/IPI GCP Install docs

https://issues.redhat.com/browse/OSDOCS-1834

Preview: 
- https://deploy-preview-33611--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned?utm_source=github&utm_campaign=bot_dp

- https://deploy-preview-33611--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-vpc.html#nw-ingress-controller-configuration-gcp-global-access_installing-gcp-vpc

Minor correction here as well:
-https://deploy-preview-33611--osdocs.netlify.app/openshift-enterprise/latest/networking/ingress-operator?utm_source=github&utm_campaign=bot_dp#nw-ingress-controller-configuration-gcp-global-access_configuring-ingress

### CODE REVIEW

**Comment by skrthomas** [comment]
@makentenza @lihongan this is ready for your review/QA. Thank you!

**Comment by makentenza** [comment]
Thanks @skrthomas.

The instructions for both sections should be the same as the cluster does not exist at that point (this is to include this configuration during IPI installation time), and the instructions from the [first link](https://deploy-preview-33611--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned?utm_source=github&utm_campaign=bot_dp#nw-ingress-controller-configuration-gcp-global-access_installing-restricted-networks-gcp-installer-provisioned) are assuming the cluster has been installed already. There is also a typo there: 

`$ oc -n openshift-ingress operator edit svc/router-default -o yaml`

Should be:

`$ oc -n openshift-ingress edit svc/router-default -o yaml`

For the [second link](https://deploy-preview-33611--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-vpc.html#nw-ingress-controller-configuration-gcp-global-access_installing-gcp-vpc) there are some bits missing:

- There is a requirement for creating the manifests, as the user will need to modify one of them. Check [this](https://docs.openshift.com/container-platform/4.7/installing/installing_aws/installing-aws-network-customizations.html#nw-aws-nlb-new-cluster_installing-aws-network-customizations) as an example for the instructions we are adding in the documentation for a similar process.
- Once the manifests are created, the file that needs to be changed is `<installation_directory>/manifests/cluster-ingress-default-ingresscontroller.yaml` and not `<installation_directory>/install-config/cluster-ingress-default-ingresscontroller.yam`
- Remove the `2. Run the following command to verify that the service allows global access:` part as there is nothing to check at that point (the cluster has not been installed yet) or add a note that the check must be done once the cluster has been provisioned to validate the service has been created with the expected annotation.

**Review by makentenza — CHANGES_REQUESTED** [review]
Please check my comments.

**Comment by skrthomas** [comment]
@makentenza Ok, I think I misunderstood the scope of your request. I now understand that this procedure needs to include the instructions for creating a manifest then editing the manifest accordingly. In my previous commit, I was trying to re-use the content from the Network Edge global Access documentation [here](https://deploy-preview-33611--osdocs.netlify.app/openshift-enterprise/latest/networking/ingress-operator.html#nw-ingress-controller-configuration-gcp-global-access_configuring-ingress)), so I had a lot of conditional text. But now that I think I'm understanding your request, I decided its better to create a whole new file with this procedure. So apologies for the delay as I was setting up another file.

Please have another look and lmk your feedback.

**Comment by makentenza** [comment]
Thanks @skrthomas,

Step 2 (Create a file that is named cluster-ingress-default-ingresscontroller.yaml in the <installation_directory>/manifests/ directory:) is not required as this is automatically created alongside the other manifests when the user executes `./openshift-install create manifests --dir=<installation_directory>`.

Note that the original content from [Network Edge global Access documentation](https://deploy-preview-33611--osdocs.netlify.app/openshift-enterprise/latest/networking/ingress-operator.html#nw-ingress-controller-configuration-gcp-global-access_configuring-ingress) still has a typo, the one I mentioned in my previous comment:

```$ oc -n openshift-ingress operator edit svc/router-default -o yaml```

Should be:

```$ oc -n openshift-ingress edit svc/router-default -o yaml```

**Comment by skrthomas** [comment]
@makentenza Ok gotcha. I removed that portion of step 2, but I left the information about "several network configuration files are in the manifests/ directory".

As for the typo in the original documentation, thanks for bringing this back to my attention. Since this doc has already passed QA, I am running your feedback by @lihongan for validation.

File: modules/nw-gcp-installing-global-access-configuration.adoc [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+[source,yaml]
+----
+  spec:
+    endpointPublishingStrategy:
+      loadBalancer:
+        providerParameters:
+          gcp:
+            clientAccess: Global <1>
+          type: GCP
+        scope: Internal
```
Comment (sgreene570) [inline modules/nw-gcp-installing-global-access-configuration.adoc:53]: Might want to take the opportunity to call out the fact that global access is only applicable to internal scope load balancers with a <2> callout here. Thoughts?

Comment (makentenza) [reply modules/nw-gcp-installing-global-access-configuration.adoc:53]: +1

**Comment by ahardin-rh** [comment]
LGTM

File: modules/nw-gcp-installing-global-access-configuration.adoc (hunk 1 above)
Comment (skrthomas) [reply modules/nw-gcp-installing-global-access-configuration.adoc:53]: @sgreene570 agree, i will add that callout. ty!! :)

Comment (sgreene570) [reply modules/nw-gcp-installing-global-access-configuration.adoc]: Fix looks good!

**Comment by makentenza** [comment]
@skrthomas changes in https://deploy-preview-33611--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned?utm_source=github&utm_campaign=bot_dp and https://deploy-preview-33611--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-vpc.html#nw-ingress-controller-configuration-gcp-global-access_installing-gcp-vpc LGTM now.

The typo from the other link is still there, are you addressing it on this PR?

**Comment by skrthomas** [comment]
@makentenza ty!! 

yep, I am just going to go ahead and put it on this PR. I added a hyphen so now the namespace is `openshift-ingress-operator`. I think that should be right bc I use the same namespace earlier in the procedure, not `openshift-ingress`. What do you think? Should both be `openshift-ingress`?

**Comment by makentenza** [comment]
@skrthomas no, the instructions earlier are ok. The `ingresscontroller/default` object is in the `openshift-ingress-operator` namespace. The `svc/router-default` object is in the `openshift-ingress` namespace. The change should be as I explained before:

`$ oc -n openshift-ingress operator edit svc/router-default -o yaml`

Should be:

`$ oc -n openshift-ingress edit svc/router-default -o yaml`

**Pull request merged**